### ISSUE TITLE
Allow a default buffer value in the buffer with readLine()

### ISSUE THREAD

**Issue by bdw429s** [jline/jline2#177]
In CommandBox, our Jline2-based CLI and package manager tool for CFML, we will often have the need to ask the user for additional input when running a "installer wizard" style command.

```bash
CommandBox > addSite
Enter webroot path: _
```

We'd like to be able to pre-populate the buffer with default text the user can edit if they wish, or just hit enter to submit it as is:

```bash
CommandBox > addSite
Enter webroot path: C:\path\to\wwwroot_
```

`readLine()` lets me set the prompt, but not anything else.  There is putString() but it's a catch 22 because you can't call putString() before readLine() because it will get output before the prompt and you can't call it after because readLine blocks until it receives your input.

For what it's worth I'm able to trick it by threading the readline() call, waiting 100ms, then outputting the default value and flushing the buffer, then joining  back to the thread to await your input. Of course, this is pretty hacky so it would be nice to see this be core.  Here's an example of my hack described above (in CFML):

```javascript
reader = shell.getReader();

var threadname = createUUID();
variables.char = '';
thread name="#threadname#" action="run" { 
  variables.char = reader.readLine( 'What is your quest? > ' );
}

sleep(100);
reader.putString( 'To seek the Holy Grail.' );
reader.flush();

thread action="join" name="#threadname#"; 

print.boldRedLine( variables.char ).toConsole();
shell.setPrompt();
```

That outputs the following, with everything after the ">" as editable:

```bash
What is your quest? > To seek the Holy Grail._
```

**Comment by bdw429s** [comment]
Thanks @gnodet !

**Comment by bdw429s** [comment]
@gnodet  I got a chance to test this out today and I'm afraid it's not quite working for me.  The buffer text is appended to the buffer, but it's not flushed to the screen so you can't actually see it.  Furthermore, if i hit the backspace key, it just erases that many characters out of my prompt.

**Comment by mvmn** [comment]
It's 2016 and JLine still doesn't support default value in input buffer? For shame.
I see that issue is fixed, but there's no release - supposed to be in 2.14.2, right? Please make the release happen soon! Can't wait for this thing to **finally** work.

**Comment by bdw429s** [comment]
@mvmn I'm not sure what release that fix made it into, but I can confirm it works in the `2.15-SNAPSHOT` on sonatype.
https://oss.sonatype.org/content/repositories/snapshots/jline/jline/2.15-SNAPSHOT/

**Comment by mvmn** [comment]
Looks like they've [released 2.14.2 on 31-st of May](https://repo1.maven.org/maven2/jline/jline/2.14.2/). 6 days after my comment.
Good.
